src/client.ts

```diff
@@ -319,26 +331,31 @@ export class Client {
 	 * such as Node.js, the `fetch` option must be provided as part of the
 	 * `options` parameter.
 	 *
-	 * @param endpoint - The Prismic REST API V2 endpoint for the repository (use
-	 *   `prismic.getEndpoint` to get the default endpoint).
+	 * @param repositoryNameOrEndpoint - The Prismic repository name or full Rest
+	 *   API V2 endpoint for the repository.
 	 * @param options - Configuration that determines how content will be queried
 	 *   from the Prismic repository.
 	 *
 	 * @returns A client that can query content from the repository.
 	 */
-	constructor(endpoint: string, options: ClientConfig = {}) {
+	constructor(repositoryNameOrEndpoint: string, options: ClientConfig = {}) {
 		if (
 			process.env.NODE_ENV === "development" &&
-			/\.prismic\.io\/(?!api\/v2\/?)/.test(endpoint)
+			/\.prismic\.io\/(?!api\/v2\/?)/.test(repositoryNameOrEndpoint)
 		) {
 			throw new PrismicError(
-				"@prismicio/client only supports Prismic Rest API V2. Please use the getEndpoint helper to generate a valid Rest API V2 endpoint URL.",
+				"@prismicio/client only supports Prismic Rest API V2. Please use the getRepositoryEndpoint helper to generate a valid Rest API V2 endpoint URL.",
 				undefined,
 				undefined,
 			);
 		}
```

Comment:
While it works like that, I'd maybe move this check inside the first `if` statement below to help reading the function (as one unfamiliar with RegExp might be wondering if this is a bug to check to `/api/v2` on something we don't know yet if an endpoint and not a name)

Reply:
Thanks, good catch